code/modules/reagents/reagent_dispenser.dm

```diff
@@ -390,4 +390,105 @@
 /obj/structure/reagent_dispensers/cauldron/can_transfer(var/obj/item/weapon/reagent_containers/R, var/mob/user)
 	if(user.a_intent != I_HELP)
 		return TRUE
-	return FALSE
\ No newline at end of file
+	return FALSE
+
+// BARRELS AND BARREL ACCESSORIES //
+/obj/structure/reagent_dispensers/cauldron/barrel
+	name = "metal barrel"
+	icon_state = "metalbarrel"
+	desc = "Originally used to store liquids & powder. It is now used as a source of comfort. This one is made of metal."
+	layer = TABLE_LAYER
+	flags = FPRINT | TWOHANDABLE | MUSTTWOHAND // If I end up being coherent enough to make it holdable in-hand
+	throwforce = 40 // Ends up dealing 20~ brute when thrown because thank you, based throw damage formula
+	var/list/exiting = list() // Manages people leaving the barrel
+
+/obj/structure/reagent_dispensers/cauldron/barrel/wood
+	name = "wooden barrel"
+	icon_state = "woodenbarrel"
+	desc = "Originally used to store liquids & powder. It is now used as a source of comfort. This one is made of wood."
+
+/obj/structure/reagent_dispensers/cauldron/barrel/update_icon()
+	return
+
+/obj/structure/reagent_dispensers/cauldron/barrel/kick_act(mob/living/carbon/human/H)
+	..()
+	if (!reagents)
+		return 1
+	if(reagents.total_volume > 10) //Beakersplashing only likes to do this sound when over 10 units
+		playsound(src, 'sound/effects/slosh.ogg', 25, 1)
+	H.investigation_log(I_CHEMS, "has emptied \a [src] ([type]) containing [reagents.get_reagent_ids(1)] onto \the [usr.loc].")
+	reagents.reaction(usr.loc)
+	src.reagents.clear_reagents()
+	H.visible_message("<span class='warning'>[usr] kicks \the [src]!</span>", "<span class='notice'>You kick \the [src].</span>")
+	for(var/atom/movable/AM in src)
+		AM.forceMove(loc)
+
+/obj/structure/reagent_dispensers/cauldron/barrel/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(iswrench(W))
+		return
+	if(istype(W,/obj/item/weapon/grab))
+		var/obj/item/weapon/grab/G = W
+		var/mob/living/target = G.affecting
+		user.visible_message("<span class='danger'>[user] begins to drag [target] into the barrel!</span>")
+		if(do_after_many(user,list(target,src),10)) //Twice the normal time
+			enter_barrel(target)
+	else
+		..()
+	return ..()
```

Comment:
Is there any reason to call `..()`? If I'm reading this correctly you'd be calling `..()` twice in the case the above _if clause_ fails

Reply:
I copy-pasta'd some code and now I feel bad. That said, I took off the default reutn ..() so help me god if I fucked up everything.